core/silkworm/common/util.cpp

```diff
@@ -154,36 +154,69 @@ std::optional<Bytes> from_hex(std::string_view hex) noexcept {
     return out;
 }
 
+inline bool case_insensitive_char_comparer(char a, char b) {
+    return(tolower(a) == tolower(b));
+}
+
+bool iequals(const std::string& a, const std::string& b) {
+    return (a.size() == b.size() && std::equal(a.begin(), a.end(), b.begin(), case_insensitive_char_comparer));
+}
+
 std::optional<uint64_t> parse_size(const std::string& sizestr) {
+
     if (sizestr.empty()) {
-        return 0;
+        return 0ull;
     }
 
-    static const std::regex pattern{"^([0-9]{1,})([\\ ]{0,})?(B|KB|MB|GB|TB)?$"};
+    static const std::regex pattern{"^(\\d*)(\\.\\d{1,3})?\\ *?(B|KB|MB|GB|TB)?$", std::regex_constants::icase};
```

Comment:
It's probably better to use [C+11 raw strings](https://www.geeksforgeeks.org/raw-string-literal-c/) for regexes to avoid backslash doubling.

Reply:
Yep. You're right.